### PR TITLE
snap-bootstrap: mount filesystems after creation

### PR DESCRIPTION
**Pull request by mvo5**
The newly created filesystems will needed to be accessed or populated
in the installation process. Mount them here for convenience.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

### CODE REVIEW

File: cmd/snap-bootstrap/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -87,6 +89,10 @@ func Run(gadgetRoot, device string, options *Options) error {
 		return err
 	}
 
+	if err := partition.MountFilesystems(created, dirs.RunMnt); err != nil {
```
Comment (cmatsuoka) [inline cmd/snap-bootstrap/bootstrap/bootstrap.go]: Maybe we should make mounting optional (using --mount or similar) for cases where we don't need the filesystems to be mounted (e.g. an initial partitioning of a loop file like we do in the integration tests). What do you think?

Comment (pedronis) [reply cmd/snap-bootstrap/bootstrap/bootstrap.go]: I agree with this consideration

File: dirs/dirs.go [hunk 2]
```diff
@@ -357,6 +359,7 @@ func SetRootDir(rootdir string) {
 	SysfsDir = filepath.Join(rootdir, "/sys")
 
 	FeaturesDir = filepath.Join(rootdir, snappyDir, "features")
+	RunMnt = filepath.Join(rootdir, "/run/mnt")
```
Comment (pedronis) [inline dirs/dirs.go]: can we have a new line before this one, also I suppose there are quite a few places to use this now?

File: cmd/snap-bootstrap/partition/deploy.go [hunk 3]
```diff
@@ -101,3 +101,21 @@ func DeployContent(created []DeviceStructure, gadgetRoot string) error {
...
+		if part.Label == "" || !part.HasFilesystem() {
+			continue
+		}
+
+		mountpoint := filepath.Join(baseMntPoint, part.Label)
+		if err := os.MkdirAll(mountpoint, 0755); err != nil {
+			return fmt.Errorf("cannot create mountpoint: %v", err)
+		}
+		if err := sysMount(part.Node, mountpoint, part.Filesystem, 0, ""); err != nil {
+			return fmt.Errorf("cannot mount filesystem %q to %q: %v", part.Node, mountpoint, err)
```
Comment (pedronis) [inline cmd/snap-bootstrap/partition/deploy.go]: s/to/at/ afaik

File: cmd/snap-bootstrap/bootstrap/bootstrap.go (hunk 1 above)
Comment (cmatsuoka) [reply cmd/snap-bootstrap/bootstrap/bootstrap.go]: Now mounting only if `create-partitions --mount` is used.

**Pull request merged**